Docs: update Disk Management guide with the new IO configuration

monicakh opened the pull request:
## What kind of change does this PR introduce?

docs update...

monicakh committed:
Docs: update Disk Management guide with the new IO configuration

J0 committed:
fix: move Auth error codes to top level (#29254)

* fix: move error codes to top level

* fix: remove stray newlines

* fix: merge all error sections into central location

* fix: add section heading above table

loong committed:
feat: prefill org and project name for first org/project (#29281)

* feat: prefill first org name and first project name

* chore: move prefill logic to new page

---------

Co-authored-by: Alaister Young <alaister@users.noreply.github.com>

joshenlim committed:
Improve table editor pagination input field (#29245)

* Improve table editor pagination input field

* sync input value with actual page value

---------

Co-authored-by: Alaister Young <a@alaisteryoung.com>

joshenlim committed:
Chore/disk mgt UI updates (#29313)

* Add note regarding final usable IOPS

* Ensure that its GiB and not GB for disk mgt ui

* Adjust language of IOPS and throughput fields, use GiB and not TiB

* PRettier

* Add note for max throughput based on compute size

* Prettier

* Small fix

joshenlim committed:
Fix type error CSV import (#29314)

inian committed:
Update 2024-05-01-pgvector-0-7-0.mdx (#29102)

joshenlim committed:
Fix styling on SSL switch (#29320)

charislam commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+### Self-Serve Configurability
+
+You can now adjust storage type, IOPS and disk size parameters yourself through the Dashboard [here](https://supabase.com/dashboard/project/_/settings/database).
```
```suggestion
You can now adjust storage type, IOPS and disk size parameters yourself through the Dashboard [Database Settings page](https://supabase.com/dashboard/project/_/settings/database).
```

charislam commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+<Admonition type="caution">
+
+Be aware that increasing IOPS or throughput incur additional charges.
```
```suggestion
Increasing IOPS or throughput incurs additional charges.
```

charislam commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+### Understanding I/O Balance and Bursting (Smaller Instances)
+
+For smaller instances like Nano, Micro, and Small, you get I/O balance and burst capacity. These instances have limited base performance for IOPS and throughput. However, they come with burst credits that let you handle temporary spikes for 30 minutes in a day in demand by briefly exceeding their baseline performance. This means you can achieve higher performance for short bursts before reverting ...
```
```suggestion
For smaller instances like Nano, Micro, and Small, you get I/O balance and burst capacity. These instances have limited base performance for IOPS and throughput. However, they come with burst credits that let you handle temporary spikes for 30 minutes in a day on demand by briefly exceeding their baseline performance. This means you can achieve higher performance for short bursts before reverting to your baseline limits.
```

kamilogorek committed:
ref: Use local state to improve GitHub integration security (#29321)

alaister committed:
fix: github auth loading state (#29322)

charislam committed:
fix(docs): code in wrong file for react auth quickstart example (#29324)

monicakh committed:
Apply suggestions from code review

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

charislam committed:
fix(cmdk): empty state shows up when it shouldn't (#29272)

* fix(cmdk): empty state shows up when it shouldn't

Can't remember why on earth I put this in a useMemo instead of a
useEffect...

* fix(command menu): switch branch command not showing up

saltcod committed:
Chore/fix move folder 2 (#29326)

* Conditional schema for move query

* Small change

---------

Co-authored-by: Joshen Lim <joshenlimek@gmail.com>

joshenlim committed:
Chore/more disk mgt UI updates (#29333)

* Only show new disk billing UI for AWS provider projects

* Fix DiskSpaceBar autoscaling should be fixed at 90%

* Change all MiB to MB and GiB to GB

* Prettier

alaister committed:
fix: error on invalid mime types (#29337)

* fix: error on invalid mime types

* allow server to check mime type instead of client

* swap ?? for ||

* fix upload icon import

staaldraad committed:
fix: stricter validation of returnTo path (#29288)

A returnTo path with /%09/place.website would cause the browser to
redirect to place.website instead of staying on the studio.

monicakh committed:
Update README.md (#29338)

fsansalvadore committed:
homepage cleanup (#29295)

* new homepage logos marquee

* add products claim

* logos spacing

* dashboard table editor demo

* dashboard table editor demo

* dashboard recordings

* rls policies

* update hp frameworks

* add dataAPIs to homepage bento

* tweet section update

* padding

* hp products update

* new customer stories hp section

* height

* finish homepage details

* update logos

* fix database visual

* improve functions visual ux

* animate data-apis visual

* realtime visual update

* reduce-motion and update chatbase logo

* lighter heading paragraph

* improve video tabs components

* add priority to offsetted images

* video plays inline

* fix mobile tweets spacing

* smoll

* load video slightly before the component enters the viewport

* fix product pointer-events

* theme based and light weight dashboard video

* testimonials margin

* remove yellow tint from light mode recordings

* fix vitest window-matchMedia error

* fix vitest window-matchMedia error

* fix url type

* remove tweet

fsansalvadore committed:
chore: remote patterns (#29263)

* chore: move domains to removePatterns in www

* chore: move domains to removePatterns in studio

* chore: move domains to removePatterns in docs

* fix

J0 committed:
fix: update docker-compose variables for auth self-hosted (#29077)

* fix: update docker-compose variables

* fix: swap = to :

saltcod committed:
Chore/unsaved changes auth template (#29296)

Trigger unsaved changes modal on auth templates

saltcod committed:
menu pref (#29297)

* Add a preference to prevent menu from expanding

* Use FormItemLayout

* Fix bottom padding

* Padding change

* Padding change back

charislam committed:
fix(ai): restrict edge function regions (#29330)

Missed some routes when restricting edge function regions in an earlier
PR. Any edge function that calls OpenAI needs to be called from an
OpenAI-supported region.

charislam committed:
fix(docs): prevent error on /docs/reference path (#29345)

jordienr committed:
prevent use of WITH, ILIKE or wildcards (*) in logs explorer (#29282)

* prevent with

* fix layout, add with clause err

* fix log explorer err message issues and test

* fix type error

* add check for ilike adn tests

* add wildcard check

* rm consolelog

J0 committed:
fix: specify which auth0 package to import for tpa (#29342)

* fix: specify which auth0 package to import for tpa

* fix: run prettier

* fix: update import

avallete committed:
chore: replace deno deprecated unstable flag with recommended (#29316)

See: https://docs.deno.com/runtime/reference/cli/unstable_flags/\#--unstable
Related: https://github.com/supabase/cli/issues/2682

dshukertjr committed:
blog: Clarify prize for the winner of LW12 hackathon (#29307)

blog: clarify prize for the winner

avallete committed:
chore: document atomic function.enabled config flag (#29355)

Related: https://github.com/supabase/cli/pull/2688

joshenlim committed:
Chore/update rr pricing for new disk mgt (#29278)

* Update pricing details in DeployNewReplicaPanel

* Update disk mgt review modal to consider RR pricing

* Update tests

* Hide utilization bar for read replicas in new disk mgt ui

* Add guard

* Update link to docs for deploy replica panel

* Address feedback

* Ensure IOPS and throughput prices are considered for replicas

* Prettier

* Attempt to fix tests

* Attempt to fix tests

* Fix calculateDiskSizePrice for replicas

* Fixes

* Prettier

* Add ternary

* Feeeex

* Various small updates from Kevin's feedback

* smol stuff

* fixes

* Update DiskManagementPanelSchema.ts

---------

Co-authored-by: Kevin Grüneberg <k.grueneberg1994@gmail.com>

fsansalvadore committed:
fix matchMedia error (#29357)

inian committed:
Update humans.txt (#29358)

J0 committed:
fix: update http to https and pathname to include ** (#29360)

sweatybridge committed:
chore: add migration fetch section to cli docs (#29356)

bulutyerli committed:
Add caution on Update Policy section for need to create Select Policy (#28678)

* Add caution on Update Policy section for need to create Select Policy

* Update apps/docs/content/guides/database/postgres/row-level-security.mdx

---------

Co-authored-by: bulut yerli <bulut@buluts-Mac-mini.local>
Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

joshenlim committed:
Ensure that backup is always available for download in restoring, restore failed or pause failed state (#29361)

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+Disk throughput is capped at 19,000 Mbps and IOPS is capped at 80,000 IOPS for all compute add-ons.
 
-If you need consistent disk performance, choose the XL or larger compute add-on which has the same baseline and maximum disk throughput and IOPS.
+| Plan        | Max Disk Throughput | Baseline Disk Throughput | Max IOPS    | Baseline IOPS |
```
`Plan` -> `Compute add-on`?

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+For smaller instances like Nano, Micro, and Small, you get I/O balance and burst capacity. These instances have limited base performance for IOPS and throughput. However, they come with burst credits that let you handle temporary spikes for 30 minutes in a day on demand by briefly exceeding their baseline performance. This means you can achieve higher performance for short bursts before reverting ...
+
+For example, the micro plan can burst up to 2,085 Mbps for 30 minutes a day and reverts to the baseline performance of 87 Mbps. Your disk IO budget gets replenished throughout the day.
```
micro instance

We're using instance/add-on/plan interchangeably in this doc. Would be best to standardize on one - I think add-on is the one that corresponds with our Pricing structure

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+Here’s how **gp3** and **io2** storage volumes differ:
+
+**gp3 (General Purpose SSD) (default)** This is a versatile and cost-effective option, perfect for a range of everyday workloads that need moderate IOPS and throughput. It supports volume sizes from 1GB to 16TB. You get IOPS ranging from 3,000 (default) up to 16,000 (maximum), and throughput from 125 MB/s (default) to 1,000 MB/s (maximum). It’s well-suited for general tasks like boot volumes and  ...
```
Remove `boot volumes` : not relevant for users

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+### Storage Volume Types: gp3 vs. io2
+
+Here’s how **gp3** and **io2** storage volumes differ:
```
This section I'm assuming is mostly lifted from AWS' descriptions. Their descriptions are generic, and don't necessarily make sense for our users. e.g. no point mentioning boot vols when our users are just selecting their data vol type.

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+**gp3 (General Purpose SSD) (default)** This is a versatile and cost-effective option, perfect for a range of everyday workloads that need moderate IOPS and throughput. It supports volume sizes from 1GB to 16TB. You get IOPS ranging from 3,000 (default) up to 16,000 (maximum), and throughput from 125 MB/s (default) to 1,000 MB/s (maximum). It’s well-suited for general tasks like boot volumes and  ...
+
+**io2 (Provisioned IOPS SSD)** Designed for high-demand scenarios where very high IOPS and low latency are crucial. It supports volumes ranging from 4GB to 16TB. With IOPS reaching up to 256,000 (depending on your instance type) and throughput scaling with IOPS, io2 is designed for large-scale databases and mission-critical applications. While it offers top-tier performance, it does come with a h ...
```
@delgado3d what did we decide was the limit, 60TB for now? Or something lower like 30TB and we can increase as needed?

delgado3d replied:
yeah we decided 60 TB, which should still give us a bit of a buffer

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
 
+<Admonition type="information">
+
+Compute add-on changes will not cause you to lose your settings.
```
`lose your disk attributes` (use `attributes`/`settings`/`configuration` consistently to avoid confusion)

darora commented on apps/docs/content/guides/platform/compute-add-ons.mdx:
```diff
@@ -52,32 +52,81 @@ When considering compute upgrades, assess whether your bottlenecks are hardware-
+
+### Constraints
+
+- After **any** attribute change, there is a cooldown period of approximately six hours before you can make further adjustments. During this time, no changes are allowed. If you encounter throttling, you’ll need to wait until the cooldown period concludes before making additional modifications.
```
`disk attribute change`

darora reviewed (commented):
some suggestions

joshenlim committed:
Remove dupe vi mock for matchMedia (#29380)

kevcodez committed:
chore: hide compute usage section for Free Plan (#29385)

djsamseng committed:
docs: Add Query nested foreign tables throught a join table example for supabase-js (#29387)

* docs: Add Query nested foreign tables throught a join table example for supabase-js

* Update apps/docs/spec/examples/examples.yml

* Update apps/docs/spec/supabase_js_v2.yml

---------

Co-authored-by: Alaister Young <alaister@users.noreply.github.com>

joshenlim committed:
Chore/notice banner for auth smtp changes (#29382)

* Notice banner for auth smtp changes

* Small style fix

* Update feedback

Hallidayo committed:
feat: auth users search (#20362)

* auth users search uid

* update new users UI

---------

Co-authored-by: Alaister Young <a@alaisteryoung.com>

monicakh committed:
Merge branch 'monicakh/update-disk-mgmt-doc' of https://github.com/supabase/supabase into monicakh/update-disk-mgmt-doc

monicakh committed:
Resolve suggestions
